### PR TITLE
Watch later

### PR DESCRIPTION
**Pull request by xii-vi**
## Added Watch Later page & Private Auth Routes

- Watch Later Page
- Require Auth Component

### CODE REVIEW

File: src/components/requireAuth/requireAuth.jsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Navigate, Outlet, useLocation } from "react-router-dom";
+import { useAuth } from "../../context/authContext";
+
+export const RequireAuth = () => {
```
Comment (ivinayakg) [inline src/components/requireAuth/requireAuth.jsx:4]: Nice abstraction for authentication

File: src/pages/singleVideo/singleVideo.jsx [hunk 2]
```diff
@@ -1,10 +1,27 @@
...
+import { useNavigate } from "react-router-dom"
 import "./singleVideo.css"
 export const SingleVideo =()=>{
-    const {videoData} = useVideo()
+    const navigate = useNavigate();
+    const {videoData,watchLaterState:{watchLaterList}, watchLaterDispatch} = useVideo()
     const {videoId} = useParams()
+    const { authState: { userLogin, encodedToken }} = useAuth();
     const videoDetails = videoData?.find(({ _id }) => _id === videoId)
+    const watchLaterHandler =()=>{
```
Comment (ivinayakg) [inline src/pages/singleVideo/singleVideo.jsx:14]: Try to see if you can shift it to another file.

**Review by ivinayakg — APPROVED** [review]
Great work 👍

**Review by yashpatel1797 — APPROVED** [review]
code looks clean.
design is consistent
folder structure is good.

**Pull request merged**